src/EntityFramework/Core/EntityClient/DbParameterCollectionHelper.cs

```diff
@@ -292,6 +292,50 @@ private static int IndexOf(IEnumerable items, string parameterName)
             return -1;
         }
 
+        /// <inhertidoc />
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1604:ElementDocumentationMustHaveSummary")]
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1611:ElementParametersMustBeDocumented")]
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1615:ElementReturnValueMustBeDocumented")]
+        [SuppressMessage("Microsoft.Usage", "CA2201:DoNotRaiseReservedExceptionTypes")]
+        public DbParameter GetParameterByValue(string value)
+        {
+            var index = IndexOfParameterValue(InnerList, value);
+            if (index < 0)
+            {
+                return null;
+            }
+
+            return InnerList[index];
+        }
+
+        private static int IndexOfParameterValue(IEnumerable items, string parameterValue)
```

Comment:
@bengutt would you say the first check will already succeed if the parameters values are identical? I think the goal of this fix should be to take advantage of the fact that lots of the parameters generated for the query are actually **identical**, i.e. the exact same table names are passed in multiple (AFAIR at least 3) parameters, plus the schema is typically always the same or has very few different values. Unless I am missing something, matching values that are not identical but similar (i.e. that only satisfy the second check but not the first one) provides only marginal value.

I am not super concerned for the performance of this but I suspect the fix could be simpler.